### ISSUE TITLE
No more relative path connection from live-crt to archive-crt files

### ISSUE THREAD

**Issue by mtraeger** [certbot/certbot#3829]
I just generated a new certificate with certbot in Docker from quay.io/letsencrypt/letsencrypt with the command 
```
docker run -it --rm -p 80:80 -p 443:443 \
-v $(pwd)/le:/etc/letsencrypt \
-v $(pwd)/le/lib:/var/lib/letsencrypt \
-v $(pwd)/le/log:/var/log/letsencrypt \
quay.io/letsencrypt/letsencrypt \
certonly \
--standalone \
--agree-tos \
--rsa-key-size 4096 \
--email EMAILADDR \
example.com
``` 

Last time I generated new certificates, there were relative symlinks from the certs in ./live/example.com/ to ./archive/example.com/ (e.g. `cert.pem -> ../../archive/www.example.com/cert1.pem`) created. But with the newest version, there are absolute paths for the symlinks (e.g. `cert.pem -> /etc/letsencrypt/archive/sub.example.com/cert1.pem`). The problem here is the usage of the (absolute) directory structure of the container, but on my host machine, the folder with the config and cert files is stored (and used) in a different location. 

Are the absolute symlink paths desired behaviour and/ or is it possible to switch back to the old relative path behaviour (e.g. via an input param)? I assume, that ./live and ./archive cannot be placed too far away from each other. 

* Version: certbot 0.10.0.dev0
* Docker-Image: quay.io/letsencrypt/letsencrypt   latest              84aa8c51b7e1
* Ubuntu 14.04

To sum up: I would expect relative paths from the certs symlinks in ./live/* to the original files in ./archive/* but now there are absolute paths for the symlinks.

As I can examine, it could be a connection to https://github.com/certbot/certbot/pull/1483
```
self.namespace.config_dir = os.path.abspath(self.namespace.config_dir)
self.namespace.work_dir = os.path.abspath(self.namespace.work_dir)
self.namespace.logs_dir = os.path.abspath(self.namespace.logs_dir)
```

**Comment by ohemorange** [comment]
This is likely to be related to #3600, which fixes other issues we had. If you would like, you can manually edit the symlinks to be relative paths after each renewal. Alternatively, in the next version of Certbot (which you can preview by pulling master, if you're feeling risk-tolerant) you will be able to manually specify the location of the archive directory.

To do this, open the relevant renewal configuration file (probably at `/etc/letsencrypt/renewal/your_domain.com.conf`), change the value of `archive_dir`, then run `certbot update_symlinks`.

**Comment by mtraeger** [comment]
Thanks for your reply. 
Isn't https://github.com/certbot/certbot/pull/3600 just fixing tests?
For the moment, I only had the issue with generating certs - but manually editing symlinks is too much work...

I would need the certificate output in `/etc/letsencrypt/` in a relative linked and compact way (e.g. to use it in different locations like it was before, also similar to Apache's behaviour: `000-default_ssl.conf -> ../sites-available/000-default_ssl.conf`), otherwise the separated generation in Docker is quite useless for me. Second reason for this request is that I had lots of fun with absolute paths before and I'd like to prevent others from similar experience.

I had a closer look at https://github.com/certbot/certbot/pull/1483 and https://github.com/certbot/certbot/issues/836 where it looks like there was an input problem with paths from the command line. Making all paths absolute, because of this, seems like a bit overkill for me. Wouldn't it be enough (or even possible) to just convert incoming paths to absolute paths and work internally (at least inside /etc/letsencrypt) with relative paths / links? OK - With the new feature of a custom archive_dir it will not be easier to support this behaviour in the next version. I'm not that used to Python, but in Java I'm using an internal path variable and decide whether I need an absolute or relative path in the end at the specific position. Maybe a similar strategy could help here and make the config and certificates more portable between different systems and file system locations.

At the moment I'm fixing my setup with a symlink to direct the absolute paths correctly to the archive directory, but for me that does not look like a persistent fix.

**Comment by pde** [comment]
Adding this to the 0.10.0 milestone so we can figure out what we're doing before it ships.

**Comment by pde** [comment]
By default I'd agree that we should make relative symlinks, not absolute ones.
